proto/p4/config/p4info.proto

```diff
@@ -65,6 +65,40 @@ message PkgInfo {
     string url = 8;
 }
 
+// wrapping the enum in a message to avoid name collisions in C++, where "enum
+// values are siblings of their type, not children of it"
+message P4Ids {
+    // ids are allocated in such a way that it is possible based on an id to
+    // deduce the resource type (e.g. table, action, counter, ...). The
+    // most-significant byte of the 32-bit id encodes the resource type. The
+    // purpose of this enum is to define which value is used as the
+    // most-significant byte for each resource type. The P4 compiler must use
+    // these values when allocating ids for P4 objects. Other users of P4Info
+    // can refer to this enum to identify a resource type based on its id.
+    enum Prefix {
+        UNSPECIFIED = 0;
+
+        // P4 language built-ins
+        ACTION = 0x01;
+        TABLE = 0x02;
+        VALUE_SET = 0x03;
+
+        // PSA externs
+        PSA_EXTERNS_START = 0x10;
+        ACTION_PROFILE = 0x11;
+        COUNTER = 0x12;
+        METER = 0x13;
+
+        // externs for other architectures (vendor extensions)
+        OTHER_EXTERNS_START = 0x80;
+
+        // max value for an unsigned 8-bit byte
+        MAX = 0xff;
+        // requires protoc >= 3.5.0
+        // reserved 0x100 to max;
```

Comment:
This comment is a little misleading.  Any value 0x100 and up is invalid since this enum defines the most significant byte of an ID.  "Reserved" implies they could be used someday.

Reply:
Isn't the "reserved" keyword a good way to ensure that we will not use these values in the future? That being said, I don't really mind removing it as "reserved" requires a very recent version of protoc.

Reply:
My brain locked in to the comment // preceding the reserved keyword, and it didn't even register as a protoc statement.